src/services/burnerWalletService.ts

```diff
@@ -4,74 +4,58 @@ import { ethers, Wallet } from 'ethers'
 import { API } from './API'
 import safeService from './safeService'
 
-export const createSignerAndWallet = async (API: API): Promise<Wallet> => {
-  const newSigner = ethers.Wallet.createRandom()
-  const walletAddress = await API.initWallet({
-    ownerAddress: newSigner.address
-  })
-  window.localStorage.setItem(
-    `cometh-connect-${walletAddress}`,
-    newSigner.privateKey
-  )
-
-  return newSigner
+export const _setSignerLocalStorage = (
+  identifier: string,
+  privateKey: string
+): void => {
+  window.localStorage.setItem(`cometh-connect-${identifier}`, privateKey)
 }
 
-export const createSignerAndWalletForUserId = async (
-  token: string,
-  userId: string,
-  API: API
-): Promise<Wallet> => {
-  const newSigner = ethers.Wallet.createRandom()
-  window.localStorage.setItem(`cometh-connect-${userId}`, newSigner.privateKey)
-
-  await API.initWalletForUserID({
-    token,
-    ownerAddress: newSigner.address
-  })
-
-  return newSigner
+export const _getSignerLocalStorage = (identifier: string): string | null => {
+  return window.localStorage.getItem(`cometh-connect-${identifier}`)
 }
 
-export const getSigner = async (
-  API: API,
-  provider: StaticJsonRpcProvider,
-  walletAddress: string
-): Promise<Wallet> => {
-  const storagePrivateKey = window.localStorage.getItem(
-    `cometh-connect-${walletAddress}`
-  )
-  if (!storagePrivateKey)
-    throw new Error(
-      'New Domain detected. You need to add that domain as signer.'
-    )
+export const createSigner = async ({
+  API,
+  walletAddress,
+  token,
+  userId
+}: {
+  API: API
+  walletAddress?: string
+  token?: string
+  userId?: string
+}): Promise<{ signer: Wallet; walletAddress: string }> => {
+  const signer = ethers.Wallet.createRandom()
 
-  const storageSigner = new ethers.Wallet(storagePrivateKey)
+  // if import external safe wallet
+  if (walletAddress) {
```

Comment:
Why this logic is here ? Cannot be the caller to just pass an identifier.
As a burnerWalletService I should not know userID notion